# HG changeset patch
# User alta88@fixall.com
# Date 1691700122 -7200
# Parent  b010af6264273d0794411ed23ccc6fe574ecd7c3
Misc: Menu item for opening feed web page in browser option. Fixes throw on invalid value of pref rss.show.content-base.

Authored by Alta88 and contributed to the Betterbird Project.
See https://github.com/Betterbird/thunderbird-patches/blob/main/LICENSE for license details.
All uses require attribution to the Author.

diff --git a/mail/base/content/mailWindowOverlay.js b/mail/base/content/mailWindowOverlay.js
--- a/mail/base/content/mailWindowOverlay.js
+++ b/mail/base/content/mailWindowOverlay.js
@@ -673,17 +673,17 @@ function InitMessageMenu() {
   }
 
   // Initialize the Open Feed Message handler menu
   let win = getBrowser()?.ownerGlobal.window;
   if (win) {
     let index = win.FeedMessageHandler.onOpenPref;
     document
       .getElementById("menu_openFeedMessage")
-      .children[index].setAttribute("checked", true);
+      .children[index]?.setAttribute("checked", true);
   }
 
   let openRssMenu = document.getElementById("openFeedMessage");
   openRssMenu.hidden = !isFeed;
   if (winType != "mail:3pane") {
     openRssMenu.hidden = true;
   }
 
diff --git a/mail/base/content/messenger-menubar.inc.xhtml b/mail/base/content/messenger-menubar.inc.xhtml
--- a/mail/base/content/messenger-menubar.inc.xhtml
+++ b/mail/base/content/messenger-menubar.inc.xhtml
@@ -974,16 +974,22 @@
                       accesskey="&openFeedSummary.accesskey;"
                       oncommand="FeedMessageHandler.onOpenPref = 1"/>
             <menuitem id="menu_openFeedWebPageInMessagePane"
                       type="radio"
                       name="openFeedGroup"
                       label="&openFeedWebPageInMP.label;"
                       accesskey="&openFeedWebPageInMP.accesskey;"
                       oncommand="FeedMessageHandler.onOpenPref = 2"/>
+            <menuitem id="menu_openFeedWebPageInBrowser"
+                      type="radio"
+                      name="openFeedGroup"
+                      label="&openFeedWebPageInBrowser.label;"
+                      accesskey="&openFeedWebPageInBrowser.accesskey;"
+                      oncommand="FeedMessageHandler.onOpenPref = 3"/>
           </menupopup>
         </menu>
 #ifdef MAIN_WINDOW
         <menuseparator id="messageAfterOpenMsgSeparator"/>
 #endif
         <menu id="msgAttachmentMenu"
               label="&openAttachmentListCmd.label;"
               accesskey="&openAttachmentListCmd.accesskey;"
diff --git a/mail/locales/en-US/chrome/messenger/messenger.dtd b/mail/locales/en-US/chrome/messenger/messenger.dtd
--- a/mail/locales/en-US/chrome/messenger/messenger.dtd
+++ b/mail/locales/en-US/chrome/messenger/messenger.dtd
@@ -422,16 +422,19 @@
 <!ENTITY openFeedMessage1.label "When Opening Feed Messages">
 <!ENTITY openFeedMessage1.accesskey "O">
 <!ENTITY openFeedWebPage.label "Open as Web Page">
 <!ENTITY openFeedWebPage.accesskey "W">
 <!ENTITY openFeedSummary.label "Open as Summary">
 <!ENTITY openFeedSummary.accesskey "S">
 <!ENTITY openFeedWebPageInMP.label "Toggle Web Page and Summary in Message Pane">
 <!ENTITY openFeedWebPageInMP.accesskey "T">
+<!ENTITY openFeedWebPageInBrowser.label "Open Web Page in Browser">
+<!ENTITY openFeedWebPageInBrowser.accesskey "B">
+
 
 <!-- Windows Menu -->
 <!ENTITY windowMenu.label "Window">
 
 <!-- Tools Menu -->
 <!ENTITY tasksMenu.label "Tools">
 <!ENTITY tasksMenu.accesskey "T">
 <!ENTITY messengerCmd.label "Mail &amp; Newsgroups">
